consul/resource_consul_node.go

```diff
@@ -34,6 +34,12 @@ func resourceConsulNode() *schema.Resource {
 				ForceNew: true,
 			},
 
+			"node_meta": &schema.Schema{
+				Type:     schema.TypeMap,
+				Optional: true,
+				ForceNew: true,
```

Comment:
I don't think this should be `ForceNew`. I believe node meta can be updated. This resource doesn't support updating though, currently. Thoughts?

Reply:
I'm not too familiar with the internals of Terraform. As far as I can tell, there is no update mechanism to catalog registration, so every change is basically a re-create.

Reply:
Registering Nodes in the catalog is an HTTP `PUT` operation, so in theory here we should be safe to remove `ForceNew` from `node_meta` (as well as `address` and `name`, I think). Both `Create` and `Update` can point to the same existing method as we have now, `resourceConsulNodeCreate`. 

This only works because the Consul API library we're using uses a `string` for the `ID` and not `*string`, meaning when we initially register the node, we send an empty string as the UUID for the node. This resource does not record that UUID and so can't re-send it as you would expect to update something with `PUT`, but since it's an empty string it matches anyway.

Reply:
There is some other things going on here too, because there is no UUID being used. If `ForceNew` is removed, we need to make sure we read `node_meta` into state in the `resourceConsulNodeRead` method, to detect any change

Reply:
I've removed the `ForceNew` and added a test for updating.

> This only works because the Consul API library we're using uses a string for the ID and not *string, meaning when we initially register the node, we send an empty string as the UUID for the node. This resource does not record that UUID and so can't re-send it as you would expect to update something with PUT, but since it's an empty string it matches anyway.

Sounds pretty ripe for future problems. Do we need to update this behaviour?